libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -224,6 +224,11 @@ size_t WiFiClient::write(const uint8_t *buf, size_t size)
     return res;
 }
 
+size_t WiFiClient::write_P(PGM_P buf, size_t size)
+{
+    return write(buf, size);
+}
+
```

Comment:
How about we create a utility mixin class, EnableProgmemMethods. This class would implement _P versions based on non-_P versions. WiFiClient would be derived from EnableProgmemMethods<WiFiClient>.

This way:
- code of WiFiClient is kept concise
- link time dead code elimination guarantees that unused methods will not end up in the final binary.